core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/spark/SparkIngestIngestDriver.java

```diff
@@ -0,0 +1,717 @@
+					.config("spark.driver.host", sparkOptions.getHost())
+					.config("spark.jars", jar)
+					.getOrCreate();
+
+			jsc = new JavaSparkContext(session.sparkContext());
+		}
+
+		// If input path is S3
+		if (basePath.startsWith("s3://")) {
+			
+			String s3EndpointUrl = ConfigAWSCommand.getS3Url(configProperties);
+			
+			Path inpuPath = (S3Path) setUpS3Filesystem(configProperties, basePath,s3EndpointUrl);
+			if (inpuPath == null || !inpuPath.isAbsolute()) {
+				LOGGER.error("Error in accessing S3 Input path " + basePath);
+				close(jsc,session);
+				return false;
+			}
+			
+			List<Path> inputFileList = new ArrayList<Path>();
+			Files.walkFileTree(inpuPath, new SimpleFileVisitor<Path>() {
+			    
+			    @Override
+			    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+			      inputFileList.add(file);
+			      return FileVisitResult.CONTINUE;
+			    }
+			  });
+		
+			int numInputFiles  = inputFileList.size();
+	
+			if (sparkOptions.getNumExecutors() == 1) {
+				 numExecutors = (int) Math.ceil((double) numInputFiles / 8);
+			}
+			else{
+				numExecutors = sparkOptions.getNumExecutors();
+			}
+			
+			if (sparkOptions.getNumCores() == 1) {
+				 numCores = 4;
+			}
+			else{
+				numCores = sparkOptions.getNumCores();
+			}
+			
+			jsc.sc().conf().set("spark.executor.instances", Integer.toString(numExecutors));
+			jsc.sc().conf().set("spark.executor.cores", Integer.toString(numCores));
+		    numPartitions = numExecutors * numCores * 2;
+		    
+			JavaRDD<URI> FileRDD = jsc.parallelize(Lists.transform(
+					inputFileList, new Function<Path, URI>() {
+
+						@Override
+						public URI apply(Path arg0) {
+							return arg0.toUri();
+						}
+					}),numPartitions);
+			
+			FileRDD.foreachPartition(uri -> {
+			
+				S3FileSystem fs = initializeS3FS(s3EndpointUrl);
+				List<URI> inputFiles  = new ArrayList<URI>();
+				while(uri.hasNext()){
+					Path inputFile = (S3Path) fs.getPath(uri.next().toString().replaceFirst(s3EndpointUrl,""));
+					inputFiles.add(inputFile.toUri());
+				}
+				
+				processInput( configFile,localInput,inputStoreName, indexList, ingestOptions,configProperties,inputFiles.iterator());	
+			});
+		}
+		// If input path is HDFS
+		else if (basePath.startsWith("hdfs://")) {		
+	
+			String hdfsFSUrl = ConfigHDFSCommand.getHdfsUrl(configProperties);
+			String hdfsInputPath = basePath.replaceFirst(
+					"hdfs://",
+					"/");
+
+			Path inpuPath = null;
+			try {
+
+				URI uri = new URI(
+						hdfsFSUrl + hdfsInputPath);
+				inpuPath = Paths.get(uri);
+				if (!Files.exists(inpuPath)) {
+					LOGGER.error("Input path " + basePath + " does not exist");
+					close(jsc,session);
+					return false;
+				}
+			}
+			catch (URISyntaxException e) {
+				LOGGER.error(
+						"Unable to ingest data, Inavlid HDFS Path",
+						e);
+				close(jsc,session);
+				return false;
+			}
+
+			List<Path> inputFileList = new ArrayList<Path>();
```

Comment:
here to line 253 is reusable between s3 and hdfs and should be outside the if/else block